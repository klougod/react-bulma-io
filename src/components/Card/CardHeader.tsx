import React from 'react'
import PropTypes from 'prop-types'

export const CardHeader = ({ className, children }: any) => (
  <header className={`card-header ${className}`}>{children}</header>
)

CardHeader.defaultProps = {
  className: ''
}

CardHeader.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node
}

export default CardHeader
