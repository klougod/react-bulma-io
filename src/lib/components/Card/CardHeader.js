import React from 'react'
import PropTypes from 'prop-types'

export const CardHeader = ({ className, children }) => (
  <header className={`card-header ${className}`}>
    { children }
  </header>
)

CardHeader.defaultProps = {
  className: ''
}

CardHeader.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default CardHeader