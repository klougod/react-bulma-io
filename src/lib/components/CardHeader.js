import React from 'react'
import PropTypes from 'prop-types'

export const CardHeader = props => {

  const {
    className,
    children
  } = props

  return (
    <header className={`card-header ${className}`}>
      { children }
    </header>
  )
}

CardHeader.defaultProps = {
  className: ''
}

CardHeader.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default CardHeader