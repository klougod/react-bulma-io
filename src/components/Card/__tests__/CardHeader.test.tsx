import React from 'react'
import Enzyme from 'enzyme'

import CardHeader from '../CardHeader'

describe('CardHeader component test', function () {
  it('should render without crash without props', function () {
    const cardHeaderMock = Enzyme.shallow(<CardHeader />)
    expect(cardHeaderMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const cardHeaderMockProps = {
      className: 'cardHeader CLass'
    }
    const cardHeaderMock = Enzyme.shallow(<CardHeader {...cardHeaderMockProps} />)
    expect(cardHeaderMock).toMatchSnapshot()
  })
})
