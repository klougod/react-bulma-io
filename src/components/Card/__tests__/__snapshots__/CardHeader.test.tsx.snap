// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader component test should render without crash with props 1`] = `
<header
  className="card-header cardHeader CLass"
/>
`;

exports[`CardHeader component test should render without crash without props 1`] = `
<header
  className="card-header "
/>
`;
